main/src/server.cpp

```diff
@@ -0,0 +1,66 @@
+#include "server.hpp"
+
+namespace server
+{
+    Server::Server(const std::string& ip, uint16_t port) noexcept :
+        m_listClient(), m_loopThread(nullptr), m_loop(uvw::Loop::getDefault())
+        {   
+            std::shared_ptr<uvw::TCPHandle> tcp = m_loop->resource<uvw::TCPHandle>();
+
+            m_loopThread = std::make_unique<std::thread>([this]() { m_loop->run(); });
+
+            tcp->on<uvw::ErrorEvent>([](const uvw::ErrorEvent &, uvw::TCPHandle &) 
+            {
+                std::cout << "An error as occured" << std::endl;
+            });
+
+            tcp->on<uvw::ListenEvent>([this](const uvw::ListenEvent &, uvw::TCPHandle &srv) {
+                std::shared_ptr<uvw::TCPHandle> client = srv.loop().resource<uvw::TCPHandle>();
+
+                client->on<uvw::CloseEvent>([ptr = srv.shared_from_this(), this](const uvw::CloseEvent &, uvw::TCPHandle &) 
+                {
+                    ptr->close();
+                    auto res = std::remove_if(m_listClient.begin(), m_listClient.end(), [](std::shared_ptr<uvw::TCPHandle> cli) { 
+                        return (cli == nullptr || cli->closing());
+                    });
+                });
+                
+                client->on<uvw::EndEvent>([this](const uvw::EndEvent &, uvw::TCPHandle &client) 
+                {
+                    client.close();
+                    auto res = std::remove_if(m_listClient.begin(), m_listClient.end(), [](std::shared_ptr<uvw::TCPHandle> cli) { 
+                        return (cli == nullptr || cli->closing());
+                    });
+                });
+                srv.accept(*client);
+                client->read();
+                m_listClient.push_back(client);
+            });
+        
+            tcp->bind(ip, port);
+            tcp->listen();
+        }
+
+        Server::~Server() noexcept
+        {
+            std::for_each(m_listClient.begin(), m_listClient.end(),
+                [this](std::shared_ptr<uvw::TCPHandle> client) {
+                    client->stop();
+                    client->clear();
+                    client->close();
+                }
+            );
+            m_listClient.clear();
+        }
+
+    void Server::Send(uint8_t* ptr, int SIZE) noexcept
+    {
+        char* pointeur = reinterpret_cast<char*>(ptr);
+        if(!m_listClient.empty())
+        {
+            std::for_each(m_listClient.begin(), m_listClient.end(),[pointeur, SIZE](std::shared_ptr<uvw::TCPHandle> client) {
+                client->write(pointeur, SIZE);
```

Comment:
should add a check to ensure that client is not nullptr and is not closed or closing